### PR TITLE
Add Bookmark module

### PR DESCRIPTION
**Pull request by Fobhep**
This is WIP - the module is "done" - however I cannot get the test running yet.
Will keep trying ;)

### CODE REVIEW

**Comment by Fobhep** [comment]
Test problem is fixed. still missing taxonomy functionality and redesign of tests

File: modules/foreman_bookmark.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    'public': 'public',
+    'query': 'query',
+}
+
+
+def main():
+    module = ForemanEntityAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            controller=dict(),
```
Comment (sean797) [inline modules/foreman_bookmark.py]: Docs say this is required?

File: modules/foreman_bookmark.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            public=dict(defaut='true', type='bool'),
+            query=dict(),
+            state=dict(default='present', choices=[
+                       'present_with_defaults', 'present', 'absent']),
+        ),
+        required_if=(
+            ['state', 'present', ['name', 'query', 'controller']],
+            ['state', 'present_with_defaults', ['name', 'query', 'controller']],
+            ['state', 'absent', ['name', 'controller']],
```
Comment (sean797) [inline modules/foreman_bookmark.py]: You can remove `name` as it required anyway. Maybe also controller, I'm not sure if that is actually required.

File: modules/foreman_bookmark.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
```
Comment (sean797) [inline modules/foreman_bookmark.py]: Would you mind moving this import below? Just so all the imports are together

File: modules/foreman_bookmark.py (hunk 1 above)
Comment (Fobhep) [reply modules/foreman_bookmark.py]: true, I will fix that here

File: modules/foreman_bookmark.py (hunk 2 above)
Comment (Fobhep) [reply modules/foreman_bookmark.py]: You can create bookmarks with identical names. In order to be unique a bookmark needs for specification both controller and name

File: modules/foreman_bookmark.py (hunk 3 above)
Comment (Fobhep) [reply modules/foreman_bookmark.py]: not at all :)

File: modules/foreman_bookmark.py (hunk 1 above)
Comment (Fobhep) [reply modules/foreman_bookmark.py]: fixed

File: modules/foreman_bookmark.py (hunk 2 above)
Comment (Fobhep) [reply modules/foreman_bookmark.py]: fixed

File: modules/foreman_bookmark.py (hunk 3 above)
Comment (Fobhep) [reply modules/foreman_bookmark.py]: fixed

**Pull request merged**